ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -316,8 +316,28 @@ extension SignalProducerType {
 	///
 	/// Returns a Disposable which can be used to interrupt the work associated
 	/// with the Signal, and prevent any future callbacks from being invoked.
-	public func start(error error: (E -> ())? = nil, completed: (() -> ())? = nil, interrupted: (() -> ())? = nil, next: (T -> ())? = nil) -> Disposable {
-		return start(Event.sink(next: next, error: error, completed: completed, interrupted: interrupted))
+	public func startWithNext(error error: (E -> ())? = nil, completed: (() -> ())? = nil, interrupted: (() -> ())? = nil, next: (T -> ())? = nil) -> Disposable {
```

Comment:
I didn't realize that the `next` param was going to be optional and have to be last. This is what I meant if the trailing closure syntax would still work in the case where it was first but the only non-optional param. If that's not possible, I don't think this is the right change and is potentially more confusing. At this point it may be back to the individual `startWithNext`, `startWithError`, etc. approach and no optional params.